app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -741,7 +751,23 @@ public void toggleMoreResultsBox() {
 
     @OnClick({R.id.inputAppraisalExpandBox})
     public void toggleAppraisalBox() {
-        toggleVisibility(inputAppraisalExpandBox, appraisalBox);
+        boolean visible = toggleVisibility(inputAppraisalExpandBox, appraisalBox);
+        moveInfoWindow(visible);
+    }
+
+    /**
+     * Move infoLayout window to top/bottom.
+     * @param toTop
+     */
+    private void moveInfoWindow(boolean toTop) {
+        if (toTop) {
+            layoutParams.gravity = Gravity.CENTER | Gravity.TOP;
+            Log.d(TAG, "moveInfoWindow: top");
+        } else {
+            layoutParams.gravity = Gravity.CENTER | Gravity.BOTTOM;
+            Log.d(TAG, "moveInfoWindow: bottom");
+        }
+        windowManager.updateViewLayout(infoLayout, layoutParams);
```

Comment:
> This... actually looks good. Is there a need to shorten phrases? I'll try merging the rest.

Short phrases, no, not really. But you would need to preserve the first option in each group. This need to be something meaningful (as there are no labels).